src/task_change.rs

```diff
@@ -374,6 +375,19 @@ fn cmp_tasks_3way(from: &Task, left: &Task, right: &Task) -> std::cmp::Ordering
     }
 }
 
+fn preferred_task_ids(t: &Task, tasks: &Vec<Task>, allowed_divergence: usize) -> Vec<usize> {
+    let mut admissibles = tasks.iter()
+        .enumerate()
+        .filter(|(_, x)| is_task_admissible(&t, &x, allowed_divergence))
+        .collect::<Vec<_>>();
+
+    admissibles.sort_unstable_by(|(_, left), (_, right)| cmp_tasks_3way(&t, &left, &right));
+
+    admissibles.iter()
```

Comment:
I think this could be `into_iter()` (and this would likely remove the need for the `*` in `*i` below?)